### PR TITLE
AutoAWQQuantizer: Option to save in hf format, Phi3: Finetune-AWQ-MB workflow

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
**AutoAWQQuantizer**
`AutoAWQQuantizer` currently saved the whole model object. However, the user might want to get the model in the standard huggingface format (config with quantization config + quantized weights). For example, genai model builder expects this format. 
- Only save the whole model object if `pack_model_for_onnx_conversion` is `True` which indicates that the model will be export to onnx using `torch.export` subsequently. Otherwise, it will be saved in the hf model format.
- Fix the io config inheritance from hf model to pytorch model.

**MergeAdapterWeights**
- Support merging weights for models finetuned using QLoRA. The quantization method used during finetuning is irrelevant for this pass. Even the original implementation merges the adapters into the float master weights 

**Phi3**
- Add option to do `AWQ` quantization on the base or fine-tuned model.
- Refactored the template so that all of the information is in one file. 
- Output of the script is also cleaned up

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: examples/phi3/phi3.py [hunk 1]
```diff
@@ -58,6 +78,11 @@
             "qlora finetuned model cannot be converted to onnx by model builder."
         ),
     )
+    parser.add_argument(
+        "--awq",
+        action="store_true",
+        help="Run AWQ on the base model or the finetuned model",
+    ),
```
[the remark this answers is not in the document]
Comment (jambayk) [reply examples/phi3/phi3.py]: Fixed

**Review by trajepl — APPROVED** [review]

File: olive/passes/pytorch/common.py [hunk 2]
```diff
@@ -43,10 +43,30 @@ def inherit_pytorch_from_hf(
...
+        # kv cache will be handled by the kv_cache flag in io_config
+        io_config = {
+            "input_names": [i for i in hf_io_config.get("input_names", []) if not i.startswith("past_key_values")],
+            "input_shapes": [],
+            "input_types": [],
+            "output_names": [o for o in hf_io_config.get("output_names", []) if not o.startswith("present")],
+            "dynamic_axes": {
+                k: v
+                for k, v in hf_io_config.get("dynamic_axes", {}).items()
+                if not k.startswith("present", "past_key_values")
```
Comment (trajepl) [inline olive/passes/pytorch/common.py:60]: Maybe you missed to convert the argument to tuple?

**Comment by trajepl** [comment]
https://github.com/microsoft/Olive/blob/main/test/unit_test/passes/pytorch/test_autoawq.py#L18

Sorry for the typos. Could you also help update the gptq to awq?

File: olive/passes/pytorch/common.py (hunk 2 above)
Comment (jambayk) [reply olive/passes/pytorch/common.py]: Done! I followed the lint error but did not see that the argument needed to be a tuple. 

Also improved the test asserts

**Review by trajepl — APPROVED** [review]

**Pull request merged**